src/core/cpu_patches.cpp

```diff
@@ -363,6 +363,8 @@ static void GenerateVCVTPH2PS(const ZydisDecodedOperand* operands, Xbyak::CodeGe
 
     SaveContext(c);
 
+    c.pushfq(); // VCVTPH2PS shouldn't modify flags
```

Comment:
I saw this in your other PR and was planning to do this when I had the chance, thanks for this. Could you move the push/popfq to Save/RestoreContext though since that's part of the general intent of those helpers.

Reply:
That's what I thought of doing initially, however what if you wanna use Save/RestoreContext in some instruction that wants to modify flags in some way and you don't want to just keep the old ones? Or is Save/Restore only going to be used on instructions that must not modify flags?

Reply:
I think having a boolean flag into Save/RestoreContext for whether to preserve flags, defaulted to false, would work.

Reply:
Done